### PR TITLE
Clean up unused gallery-partnerships code [GALL-1487]

### PR DESCRIPTION
**Pull request by mdole**
As of ~ 2 years ago, we no longer use `artsy.net/gallery-partnerships`, preferring instead to redirect users to `partners.artsy.net`. This PR cleans up some of the unused code.

Interestingly, if I'm understanding the pattern correctly, the routes that are redirecting from `/gallery-partnerships` are sort of stacked on top of routes defined in `apps/partnerships` - a lot of the cleanup in this PR is of code within that partnerships app. I left the redirect defined in `apps/gallery_partnerships` just to be safe, but we could also consider removing those as well.

@oxaudo I think you created this ticket originally; let me know if there are other pieces you think could be cleaned up!

Jira ticket: https://artsyproduct.atlassian.net/browse/GALL-1487

### CODE REVIEW

File: src/desktop/apps/partnerships/client/router.coffee [hunk 1]
```diff
@@ -5,8 +5,6 @@ Jump = require '../../../components/jump/view.coffee'
 
 module.exports = class PartnershipsRouter extends Backbone.Router
   routes:
-    'gallery-partnerships': 'toTop'
-    'gallery-partnerships/:slug': 'toSection'
     'institution-partnerships': 'toTop'
     'institution-partnerships/:slug': 'toSection'
     'auction-partnerships': 'toTop'
```
Comment (oxaudo) [inline src/desktop/apps/partnerships/client/router.coffee:10]: I know you've asked if those other pages are still used and the reply was that 'institutions' get some traffic. Wondering if 'auction-partnerships' is relevant to anything? Seems like we've covered most auction houses by now:)

But I guess it's ok to keep for now. -600 lines of code is better than nothing:)

File: src/desktop/apps/partnerships/templates/common_section_mixins.jade [hunk 2]
```diff
@@ -73,26 +69,3 @@ mixin apply(section)
     if path.indexOf('institution') > 0
```
Comment (oxaudo) [inline src/desktop/apps/partnerships/templates/common_section_mixins.jade:69]: I guess this mixin is never called for 'auction-partnerships'. Wondering if we even need the conditional here?

File: src/desktop/apps/partnerships/templates/meta.jade [hunk 3]
```diff
@@ -1,16 +1,4 @@
...
-  meta( name="og:description", content="Artsy invites the best galleries in the world to list their programs and contribute to the most comprehensive online resource for seasoned and aspiring collectors alike." )
-  meta( property="twitter:description", content="Artsy invites the best galleries in the world to list their programs and contribute to the most comprehensive online resource for seasoned and aspiring collectors alike." )
-
-  link( rel="canonical", href=(sd.APP_URL + "/gallery-partnerships") )
-  meta( property="og:url", content=(sd.APP_URL + "/gallery-partnerships") )
-  meta( property="og:image", content=asset('/images/og_image.jpg') )
-  meta( property="og:type", content="website" )
-  meta( property="twitter:card", content="summary" )
-else if path.indexOf('institution') > 0
+if path.indexOf('institution') > 0
```
Comment (oxaudo) [inline src/desktop/apps/partnerships/templates/meta.jade:1]: Again - seems like it's only called for institutions now. Maybe we don't even need the conditional checks now?

File: src/desktop/apps/partnerships/client/router.coffee (hunk 1 above)
Comment (mdole) [reply src/desktop/apps/partnerships/client/router.coffee:10]: yeah, apparently it is still in use - see Jacqueline's comment [here](https://artsy.slack.com/archives/C02MU3DB3/p1598466516020900) and the thread just below that for a bit more detail if you're interested. Sounds like they're planning to update the photos and copy on both of those pages

File: src/desktop/apps/partnerships/templates/common_section_mixins.jade (hunk 2 above)
Comment (mdole) [reply src/desktop/apps/partnerships/templates/common_section_mixins.jade:69]: Looks like we don't! good thought

File: src/desktop/apps/partnerships/templates/meta.jade (hunk 3 above)
Comment (mdole) [reply src/desktop/apps/partnerships/templates/meta.jade:1]: Hm if I'm reading this right, aren't we still using it for Institutions or for auctions? Or is there somewhere else that's overwriting the `Auctions` meta?

**Comment by mdole** [comment]
going to go ahead and merge this! happy to follow up with tweaks if I'm wrong about the `meta.jade` file

File: src/desktop/apps/partnerships/templates/meta.jade (hunk 3 above)
Comment (oxaudo) [reply src/desktop/apps/partnerships/templates/meta.jade:1]: Oh... have not noticed the last else below... Yep, this one makes sense.

**Pull request merged**